src/NServiceBus.NHibernate/SharedSession/SharedSessionConfig.cs

```diff
@@ -0,0 +1,21 @@
+namespace NServiceBus.Persistence.NHibernate
+{
+    using Configuration.AdvanceExtensibility;
+    using NServiceBus.Features;
+
+    /// <summary>
+    /// Deduplication configuration extensions.
+    /// </summary>
+    public static class SharedSessionConfig
+    {
+        /// <summary>
+        /// Disables sharing a connection between SQL Server transport and NHibernate persistence. Intended to be used in conjunction with Outbox only.
+        /// </summary>
+        /// <param name="persistenceConfiguration"></param>
+        public static PersistenceExtentions<NHibernatePersistence> DoNotShareTransportConnection(this PersistenceExtentions<NHibernatePersistence> persistenceConfiguration)
```

Comment:
That's partially correct. `.EnableOutbox()` disables the ambient transactions and user would get an exception when starting. On the other hand, If he calls `EnableFeature<Outbox>()` that would not disable the ambient transactions. In such case to properly handle updates user needs to use `NHibernateStorageContext` to access his data. If he creates a session himself (which e.g. is required in order to use second-level cache), that session's connection will be enlisted in `OutboxDedupBehavior` new transaction scope, not the transport one resulting in potential inconsistent operations.